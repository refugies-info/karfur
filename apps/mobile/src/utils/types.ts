import React from "react";

// Utility type to extract props type from a component
export type PropsOf<T> = T extends React.ComponentType<infer P> ? P : never;
