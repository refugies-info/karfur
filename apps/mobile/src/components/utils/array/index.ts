export { default as isLastChild } from "./isLastChild";
