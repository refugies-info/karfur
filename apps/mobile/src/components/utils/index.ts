export * from "./array";
export { isDarkColor } from "./isDarkColor";
