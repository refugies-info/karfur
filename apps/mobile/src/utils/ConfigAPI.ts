import axios from "react-native-axios";
import Config from "../libs/getEnvironment";

export const dbURL = Config.dbUrl;
export const headers = {
  "Content-Type": "application/json",
  "site-secret": Config.siteSecret || "",
};
export const apiCaller = axios.create({
  baseURL: dbURL,
  headers,
});