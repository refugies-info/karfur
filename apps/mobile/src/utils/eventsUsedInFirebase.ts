export enum FirebaseEvent {
  REACH_END_CONTENT = "REACH_END_CONTENT",
  VALIDATE_LANGUAGE = "VALIDATE_LANGUAGE",
  VALIDATE_LOCATION = "VALIDATE_LOCATION",
  VALIDATE_AGE = "VALIDATE_AGE",
  VALIDATE_FRENCH_LEVEL = "VALIDATE_FRENCH_LEVEL",
  PROFILE_COMPLETED = "PROFILE_COMPLETED",
  PROFILE_NOT_COMPLETED = "PROFILE_NOT_COMPLETED",
  CLIC_THEME = "CLIC_THEME",
  CLIC_NEED = "CLIC_NEED",
  CLIC_CONTENT = "CLIC_CONTENT",
  CLIC_LIST = "CLIC_LIST",
  CLIC_CAROUSEL = "CLIC_CAROUSEL",
  CLIC_ACCORDION_ENGAGEMENT = "CLIC_ACCORDION_ENGAGEMENT",
  CLIC_ACCORDION = "CLIC_ACCORDION",
  CLIC_SEE_WEBSITE = "CLIC_SEE_WEBSITE",
  CLIC_SEE_MAP = "CLIC_SEE_MAP",
  SHARE_CONTENT = "SHARE_CONTENT",
  ADD_FAVORITE = "ADD_FAVORITE",
  SWITCH_LANGUAGE = "SWITCH_LANGUAGE",
  LONG_PRESS_CHANGE_LANGUAGE = "LONG_PRESS_CHANGE_LANGUAGE",
  START_VOICEOVER = "START_VOICEOVER",
  OPEN_NOTIFICATION = "OPEN_NOTIFICATION",
  CLIC_NOTIFICATION_ICON = "CLIC_NOTIFICATION_ICON",
  TOGGLE_NOTIFICATION_TYPE = "TOGGLE_NOTIFICATION_TYPE",
  TOGGLE_NOTIFICATION_THEME = "TOGGLE_NOTIFICATION_THEME",
}
