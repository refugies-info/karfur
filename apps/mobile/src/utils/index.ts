export { default as invariant } from "./invariant";
export * from "./types";
