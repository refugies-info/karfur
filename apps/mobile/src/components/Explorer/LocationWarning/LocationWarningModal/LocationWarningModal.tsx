import * as React from "react";
import { Pressable, View } from "react-native";
import * as Linking from "expo-linking";
import { Icon } from "react-native-eva-icons";
import Modal from "react-native-modal";
import { StyleSheet } from "react-native";
import { RTLView } from "../../../BasicComponents";
import { styles } from "../../../../theme";
import { SmallButton } from "../../../SmallButton";
import {
  TextDSFR_XL,
  TextDSFR_MD,
  TextDSFR_MD_Bold,
} from "../../../StyledText";
import { FixSafeAreaView } from "../../../FixSafeAreaView";
import { useTranslationWithRTL } from "../../../../hooks/useTranslationWithRTL";
import Map from "../../../../theme/images/localizedWarning/france_map.svg";
import Pin from "../../../../theme/images/localizedWarning/pin_traffic_cone.svg";
import styled, { useTheme } from "styled-components/native";
import { withProps } from "../../../../utils";
import { useSafeAreaInsets } from "react-native-safe-area-context";
import { Columns, ColumnsSpacing } from "../../../layout";
import { useNavigation } from "@react-navigation/native";

interface Props {
  isVisible: boolean;
  closeModal: any;
  nbGlobalContent: number;
  nbLocalizedContent: number;
  city: string;
}

const stylesheet = StyleSheet.create({
  subtitle: {
    color: styles.colors.darkBlue,
    textAlign: "center",
  },
  centerText: {
    textAlign: "center",
  },
});

// TODO Put in icons/FloatingCloseButton ?
const CloseButton = withProps(() => {
  const { t } = useTranslationWithRTL();
  const insets = useSafeAreaInsets();
  return {
    iconName: "close-outline",
    label: t("global.close", "Fermer"),
    top: insets.top,
  };
})(styled(SmallButton)<{ top: number }>`
  position: absolute;
  top: ${({ top }) => top}px;
  ${({ theme }) => (theme.i18n.isRTL ? "left: 0" : "right: 0")};
`);

// TODO Put in icons/MonitorIcon
const MonitorIcon = withProps(({ size }: { size: number }) => {
  const theme = useTheme();
  return {
    name: "monitor-outline",
    height: size,
    width: size,
    fill: theme.colors.black,
  };
})(Icon);

export const LocationWarningModal = (props: Props) => {
  const { t } = useTranslationWithRTL();
  const navigation = useNavigation<any>();
  const goTo = () => {
    props.closeModal();
    navigation.navigate("NearMeCardsScreen");
  };

  return (
    <Modal
      isVisible={props.isVisible}
      statusBarTranslucent={true}
      backdropColor={styles.colors.greyF7}
      backdropOpacity={1}
    >
      <FixSafeAreaView style={{ flex: 1, justifyContent: "center" }}>
        <CloseButton onPress={props.closeModal} />
        <RTLView style={{ justifyContent: "center", alignItems: "flex-end" }}>
          <View>
            <Map
              width={102}
              height={104}
              style={{ marginBottom: styles.margin }}
            />
            <TextDSFR_MD_Bold style={stylesheet.subtitle}>
              {t("explorer_screen.nb_content", {
                nbContent: props.nbGlobalContent,
              })}
            </TextDSFR_MD_Bold>
            <TextDSFR_MD_Bold style={stylesheet.subtitle}>
              {t("explorer_screen.country_content")}
            </TextDSFR_MD_Bold>
          </View>
          <Icon
            name="plus-outline"
            height={24}
            width={24}
            fill={styles.colors.darkBlue}
            style={{ marginHorizontal: styles.margin * 2 }}
          />
          <Pressable
            accessibilityRole="button"
            accessible={true}
            onPress={goTo}
          >
            <Pin
              width={114}
              height={104}
              style={{ marginBottom: styles.margin }}
            />
            <TextDSFR_MD_Bold
              style={{
                ...stylesheet.subtitle,
                textDecorationLine:
                  props.nbLocalizedContent > 0 ? "underline" : "none",
              }}
            >
              {t("explorer_screen.nb_content", {
                nbContent: props.nbLocalizedContent,
              })}
            </TextDSFR_MD_Bold>
            <TextDSFR_MD_Bold style={stylesheet.subtitle}>
              {t("explorer_screen.city_content", {
                city: props.city,
              })}
            </TextDSFR_MD_Bold>
          </Pressable>
        </RTLView>

        <View style={{ marginTop: styles.margin * 5 }}>
          <TextDSFR_XL style={stylesheet.centerText}>
            {t("explorer_screen.development_in_progress")}
          </TextDSFR_XL>
          <TextDSFR_MD
            style={{
              marginVertical: styles.margin * 2,
            }}
          >
            {t("explorer_screen.adding_new_content")}
          </TextDSFR_MD>
          <Columns
            RTLBehaviour
            layout="auto"
            horizontalAlign="center"
            verticalAlign="center"
          >
            <MonitorIcon size={24} />
            <TextDSFR_MD_Bold
              onPress={() => Linking.openURL("https://www.refugies.info")}
              accessibilityRole="link"
            >
              www.refugies.info
            </TextDSFR_MD_Bold>
          </Columns>
        </View>
      </FixSafeAreaView>
    </Modal>
  );
};

export default LocationWarningModal;
