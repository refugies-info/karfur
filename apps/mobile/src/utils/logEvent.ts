/* eslint-disable no-console */
import analytics from "@react-native-firebase/analytics";
import Config from "~/libs/getEnvironment";
import { FirebaseEvent } from "./eventsUsedInFirebase";

export const logEventInFirebase = async (eventName: FirebaseEvent, data: Record<string, any>) => {
  const { envName, debugModeFirebase } = Config;
  if (envName === "DEVELOPMENT") {
    if (debugModeFirebase) {
      console.log("Environment is dev and debug mode is activated for firebase");
    } else {
      console.log("Environment is dev and debug mode is NOT activated for firebase");
      return;
    }
  }

  if (envName && ["DEVELOPMENT", "STAGING", "PROD"].includes(envName)) {
    await analytics().logEvent(eventName, { ...data, envName });
  }
};
