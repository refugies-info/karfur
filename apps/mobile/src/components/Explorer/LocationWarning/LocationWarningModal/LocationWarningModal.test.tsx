import React from "react";
import LocationWarningModal from "./LocationWarningModal";
import { render } from "../../../utils/tests";

jest.mock("react-i18next", () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

jest.mock("@react-navigation/native", () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

// @see https://stackoverflow.com/questions/50793885/referenceerror-you-are-trying-to-import-a-file-after-the-jest-environment-has
jest.useFakeTimers();

describe("LocationWarningModal snapshot test suite", () => {
  it("should render without bug", async () => {
    const test = await render(
      <LocationWarningModal
        isVisible
        nbGlobalContent={42}
        nbLocalizedContent={6}
        city="Roubaix"
        closeModal={() => null}
      />
    );
    expect(test).toMatchSnapshot();
  });
});
