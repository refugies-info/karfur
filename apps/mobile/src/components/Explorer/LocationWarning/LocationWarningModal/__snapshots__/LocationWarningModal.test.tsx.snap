// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationWarningModal snapshot test suite should render without bug 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={true}
  supportedOrientations={
    [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#F7F7F7",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    collapsable={false}
    deviceHeight={null}
    deviceWidth={null}
    hideModalContentWhileAnimating={false}
    onBackdropPress={[Function]}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    panResponderThreshold={4}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    statusBarTranslucent={true}
    style={
      {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": [
          {
            "translateY": 1334,
          },
        ],
      }
    }
    supportedOrientations={
      [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        {
          "bottom": 0,
          "left": 0,
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityLabel="global.close"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#FFFFFF",
            "borderBottomLeftRadius": 12,
            "borderBottomRightRadius": 12,
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "elevation": 6,
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
            "position": "absolute",
            "right": 0,
            "shadowColor": "#212121",
            "shadowOffset": {
              "height": 1,
              "width": 1,
            },
            "shadowOpacity": 0.16,
            "shadowRadius": 8,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="24"
            bbWidth="24"
            fill="#212121"
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={24}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4280361249,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGRect
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    height="24"
                    matrix={
                      [
                        -1,
                        1.2246467991473532e-16,
                        -1.2246467991473532e-16,
                        -1,
                        24,
                        24,
                      ]
                    }
                    opacity={0}
                    width="24"
                    x="0"
                    y="0"
                  />
                  <RNSVGPath
                    d="M13.41 12l4.3-4.29a1 1 0 1 0-1.42-1.42L12 10.59l-4.29-4.3a1 1 0 0 0-1.42 1.42l4.3 4.29-4.3 4.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0l4.29-4.3 4.29 4.3a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
      </View>
      <View
        style={
          [
            {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
            },
            {
              "alignItems": "flex-end",
              "justifyContent": "center",
            },
          ]
        }
      >
        <View>
          <SvgMock
            height={104}
            style={
              {
                "marginBottom": 8,
              }
            }
            width={102}
          />
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#212121",
                  "fontFamily": "Marianne-Bold",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                },
                {
                  "color": "#0421B1",
                  "textAlign": "center",
                },
              ]
            }
          >
            explorer_screen.nb_content
          </Text>
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#212121",
                  "fontFamily": "Marianne-Bold",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                },
                {
                  "color": "#0421B1",
                  "textAlign": "center",
                },
              ]
            }
          >
            explorer_screen.country_content
          </Text>
        </View>
        <View
          collapsable={false}
          style={
            {
              "transform": [
                {
                  "scale": 1,
                },
              ],
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="24"
            bbWidth="24"
            fill="#0421B1"
            focusable={false}
            height={24}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "marginHorizontal": 16,
                },
                {
                  "flex": 0,
                  "height": 24,
                  "width": 24,
                },
              ]
            }
            vbHeight={24}
            vbWidth={24}
            width={24}
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278460849,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                >
                  <RNSVGRect
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                    height="24"
                    matrix={
                      [
                        -1,
                        1.2246467991473532e-16,
                        -1.2246467991473532e-16,
                        -1,
                        24,
                        24,
                      ]
                    }
                    opacity={0}
                    width="24"
                    x="0"
                    y="0"
                  />
                  <RNSVGPath
                    d="M19 11h-6V5a1 1 0 0 0-2 0v6H5a1 1 0 0 0 0 2h6v6a1 1 0 0 0 2 0v-6h6a1 1 0 0 0 0-2z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <SvgMock
            height={104}
            style={
              {
                "marginBottom": 8,
              }
            }
            width={114}
          />
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#212121",
                  "fontFamily": "Marianne-Bold",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                },
                {
                  "color": "#0421B1",
                  "textAlign": "center",
                  "textDecorationLine": "underline",
                },
              ]
            }
          >
            explorer_screen.nb_content
          </Text>
          <Text
            isRTL={false}
            selectable={true}
            style={
              [
                {
                  "color": "#212121",
                  "fontFamily": "Marianne-Bold",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                },
                {
                  "color": "#0421B1",
                  "textAlign": "center",
                },
              ]
            }
          >
            explorer_screen.city_content
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "marginTop": 40,
          }
        }
      >
        <Text
          isRTL={false}
          selectable={true}
          style={
            [
              {
                "color": "#212121",
                "fontFamily": "Marianne-Bold",
                "fontSize": 24,
                "lineHeight": 32,
                "textAlign": "left",
              },
              {
                "textAlign": "center",
              },
            ]
          }
        >
          explorer_screen.development_in_progress
        </Text>
        <Text
          isRTL={false}
          selectable={true}
          style={
            [
              {
                "color": "#212121",
                "fontFamily": "Marianne-Regular",
                "fontSize": 16,
                "lineHeight": 24,
                "textAlign": "left",
              },
              {
                "marginVertical": 16,
              },
            ]
          }
        >
          explorer_screen.adding_new_content
        </Text>
        <View
          RTLBehaviour={true}
          animatedStyle={
            {
              "value": {},
            }
          }
          collapsable={false}
          horizontalAlign="center"
          style={
            {
              "alignItems": "center",
              "flexBasis": "auto",
              "flexDirection": "row",
              "flexGrow": 1,
              "justifyContent": "center",
            }
          }
          verticalAlign="center"
        >
          <View
            style={
              {
                "flexBasis": "auto",
                "flexGrow": 0,
                "marginRight": 8,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "transform": [
                    {
                      "scale": 1,
                    },
                  ],
                }
              }
            >
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="24"
                bbWidth="24"
                fill="#212121"
                focusable={false}
                height={24}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "flex": 0,
                      "height": 24,
                      "width": 24,
                    },
                  ]
                }
                vbHeight={24}
                vbWidth={24}
                width={24}
              >
                <RNSVGGroup
                  fill={
                    {
                      "payload": 4280361249,
                      "type": 0,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  >
                    <RNSVGGroup
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    >
                      <RNSVGRect
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                        height="24"
                        opacity={0}
                        width="24"
                        x="0"
                        y="0"
                      />
                      <RNSVGPath
                        d="M19 3H5a3 3 0 0 0-3 3v8a3 3 0 0 0 3 3h6v2H7a1 1 0 0 0 0 2h10a1 1 0 0 0 0-2h-4v-2h6a3 3 0 0 0 3-3V6a3 3 0 0 0-3-3zm1 11a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V6a1 1 0 0 1 1-1h14a1 1 0 0 1 1 1z"
                        fill={
                          {
                            "payload": 4278190080,
                            "type": 0,
                          }
                        }
                      />
                    </RNSVGGroup>
                  </RNSVGGroup>
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
          <View
            style={
              {
                "flexBasis": "auto",
                "flexGrow": 0,
              }
            }
          >
            <Text
              accessibilityRole="link"
              isRTL={false}
              onPress={[Function]}
              selectable={true}
              style={
                {
                  "color": "#212121",
                  "fontFamily": "Marianne-Bold",
                  "fontSize": 16,
                  "lineHeight": 24,
                  "textAlign": "left",
                }
              }
            >
              www.refugies.info
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
